File: pe-core/src/main/java/org/saar/core/renderer/deferred/shadow/ShadowRenderingOutput.java
package org.saar.core.renderer.deferred.shadow;

import org.saar.core.renderer.RenderingOutput;
import org.saar.core.screen.Screen;
import org.saar.lwjgl.opengl.textures.ReadOnlyTexture;

public class ShadowRenderingOutput implements RenderingOutput {

    private final Screen screen;

    public ShadowRenderingOutput(Screen screen) {
        this.screen = screen;
    }

    @Override
    public void to(Screen screen) {
        this.screen.copyTo(screen);
    }

    public ReadOnlyTexture getShowMap() {
        return null;
    }

}

File: pe-core/src/main/java/org/saar/core/renderer/deferred/shadow/ShadowRenderingOutput.kt
package org.saar.core.renderer.deferred.shadow

import org.saar.core.renderer.RenderingOutput
import org.saar.core.screen.Screen
import org.saar.lwjgl.opengl.textures.ReadOnlyTexture

class ShadowRenderingOutput(private val screen: Screen, private val depthTexture: ReadOnlyTexture) : RenderingOutput {

    override fun to(screen: Screen) = this.screen.copyTo(screen)

    override fun toTexture() = this.depthTexture

}